### PR TITLE
add transaction size checking before adding it to mempool

### PR DESCRIPTION
**Pull request by racs4**
related to #220

### CODE REVIEW

File: kindelia_core/src/api.rs [hunk 1]
```diff
@@ -289,7 +289,7 @@ pub struct RegInfo {
 pub type ReqAnsSend<T> = oneshot::Sender<T>;
 pub type ReqAnsRecv<T> = oneshot::Receiver<T>;
 
-type PublishResults = Vec<Result<(), ()>>;
+type PublishResults = Vec<Result<(), String>>;
```
Comment (steinerkelvin) [inline kindelia_core/src/api.rs:292]: Let's avoid doing this and start explicitly defining the error types as enums. 
e.g. `EffectfFailure`.

**Review by steinerkelvin — CHANGES_REQUESTED** [review]

File: kindelia_core/src/api.rs (hunk 1 above)
Comment (racs4) [reply kindelia_core/src/api.rs:292]: Yes, I would suggest that

Comment (dan-da) [reply kindelia_core/src/api.rs:292]: perhaps consider use of thiserror crate.

Comment (steinerkelvin) [reply kindelia_core/src/api.rs:292]: @dan-da `thiserror` seems preety useful tbh. thanks!

**Pull request merged**